### PR TITLE
Fix project import with skeletons

### PR DESCRIPTION
**Pull request by yasakova-anastasia**
### Motivation and context

Depends on https://github.com/opencv/cvat/pull/4984

- Small fixes for backups (an issue with same sublabels)
- Fixed project import with skeletons

### Checklist

- [x] I submit my changes into the `develop` branch
- [ ] I have added a description of my changes into [CHANGELOG](https://github.com/cvat-ai/cvat/blob/develop/CHANGELOG.md) file
- [x] I have updated the [documentation](
  https://github.com/cvat-ai/cvat/blob/develop/README.md#documentation) accordingly
- [x] I have added tests to cover my changes
- [ ] I have linked related issues ([read github docs](
  https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword))
- [ ] I have increased versions of npm packages if it is necessary ([cvat-canvas](https://github.com/cvat-ai/cvat/tree/develop/cvat-canvas#versioning),
[cvat-core](https://github.com/cvat-ai/cvat/tree/develop/cvat-core#versioning), [cvat-data](https://github.com/cvat-ai/cvat/tree/develop/cvat-data#versioning) and [cvat-ui](https://github.com/cvat-ai/cvat/tree/develop/cvat-ui#versioning))

### License

- [x] I submit _my code changes_ under the same [MIT License](
  https://github.com/cvat-ai/cvat/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.

### CODE REVIEW

File: cvat/requirements/base.txt [hunk 1]
```diff
@@ -47,7 +47,7 @@ azure-storage-blob==12.13.0
 google-cloud-storage==1.42.0
 # --no-binary=datumaro: workaround for pip to install
 # opencv-headless instead of regular opencv, to actually run setup script
-datumaro==0.2.0 --no-binary=datumaro
+git+https://github.com/cvat-ai/datumaro@ay/labels_extension
```
Comment (kirill-sizov) [inline cvat/requirements/base.txt:50]: Update this requirement as the `ay/labels_extension` extension no longer exists.

File: cvat/apps/dataset_manager/formats/cvat.py [hunk 2]
```diff
@@ -164,7 +187,19 @@ def _parse(cls, path):
                     }
                     subset = el.attrib.get('subset')
             elif ev == 'end':
-                if el.tag == 'attribute' and attributes is not None:
+                if el.tag == 'attribute' and element_attributes is not None and shape_element is not None:
```
Comment (kirill-sizov) [inline cvat/apps/dataset_manager/formats/cvat.py:190]: When is `element_attributes` value can be `not None`? I did not find a place in the code where `element_attributes` is assigned to any value

File: cvat/requirements/base.txt (hunk 1 above)
Comment (yasakova-anastasia) [reply cvat/requirements/base.txt]: Updated. I opened a [PR](https://github.com/opencv/cvat/pull/4984) with Datumaro version update

File: cvat/apps/dataset_manager/formats/cvat.py (hunk 2 above)
Comment (yasakova-anastasia) [reply cvat/apps/dataset_manager/formats/cvat.py:190]: Yes, you're right. Fixed

File: cvat/apps/dataset_manager/formats/cvat.py [hunk 3]
```diff
@@ -118,23 +118,34 @@ def _parse(cls, path):
         items = OrderedDict()
 
         track = None
+        track_element = None
+        track_shapes = None
         shape = None
+        shape_element = None
```
[the remark this answers is not in the document]
Comment (yasakova-anastasia) [reply cvat/apps/dataset_manager/formats/cvat.py:124]: Shape element is a skeleton element. `shape_element is not None` when we have a skeleton (shape).

**Review by kirill-sizov — COMMENTED** [review]
I have one bug on my side, here is steps to reproduce.

1. Create project with skeleton
2. Create task in project with video
3. Add skeleton track
4. Go to the frame with number `last_frame - 10`
5. Mark one point of skeleton as `outside` on this frame
6. Save annotations
7. Export dataset for project (save_images = True)
8. Import exported dataset

Result: An outside property is not stored in an imported task

**Comment by yasakova-anastasia** [comment]
@kirill-sizov, thanks. Fixed.

**Comment by kirill-sizov** [comment]
Full check can be considered as green for this PR, the reason why is was red currently unknown, but I've created issue about this problem.

**Review by kirill-sizov — APPROVED** [review]
Functionality works fine (in develop environment) , code changes looks good to me.

And I want to mention two things:

1) Need to fix project export (KeyError) in separate PR;
2) Currently CVAT format it's really hard to support, this contains 1.3k lines of code, a lot of code duplication, almost 200 `if` statements. It's hard to review and difficult to extend. I think if we planing to extend this format in future with new shapes or something else we should to refactor `cvat.py` first.

File: site/content/en/docs/manual/advanced/xml_format.md [hunk 4]
```diff
@@ -41,6 +41,7 @@ In annotation mode each image tag has `width` and `height` attributes for the sa
           <attributes>
             <attribute>
               <name>String: attribute name</name>
+              <type>String: any, bbox, cuboid, cuboid_3d, ellipse, polygon, polyline, points, skeleton, tag</type>
```
Comment (nmanovic) [inline site/content/en/docs/manual/advanced/xml_format.md:44]: Is it for attributes? I believe it is a type for the label itself.

Comment (yasakova-anastasia) [reply site/content/en/docs/manual/advanced/xml_format.md]: Fixed, thanks

**Pull request merged**